deps/wazuh_testing/wazuh_testing/vulnerability_detector.py

```diff
@@ -98,6 +99,42 @@
                             "اختبار"]
 
 
+def mock_cve_db(func):
+    """Decorator used in any function that needs to mock cve.db
+
+    This function will execute `func` after stopping wazuh-modulesd and wazuh-db and cleaning the db. After that,
+    it will start the daemons again, clean the logs, etc.
+
+    Args:
+         func (callable): function that will mock the cve.db
+
+    Example:
+        @vd.mock_cve_db
+        def mock_vulnerability_scan(request, mock_agent):
+    """
+    @functools.wraps(func)
+    def magic(*args, **kwargs):
+        control_service('stop', daemon='wazuh-modulesd')
+        control_service('stop', daemon='wazuh-db')
+
+        # Clean tables
+        clean_vd_tables(agent=kwargs['mock_agent'])
+
+        func(*args, **kwargs)
+
+        control_service('start', daemon='wazuh-modulesd')
+        control_service('start', daemon='wazuh-db')
+
+        # Truncate ossec.log
+        file.truncate_file(LOG_FILE_PATH)
```

Comment:
Maybe it would be a better idea to truncate the file BEFORE starting the modules, to avoid required logs appearing before the truncation and tests failing for this reason.

Reply:
Yep, you're right. I just pushed a commit doing applying that change. 

```
[root@centos-8 test_vulnerability_detector]# python3 -m pytest test_scan_results
============================================================================================== test session starts ==============================================================================================
platform linux -- Python 3.6.8, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /tmp/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.9.0, html-2.0.1, testinfra-5.0.0
collected 47 items

test_scan_results/test_debian_inventory_debian_feed.py ..                                                                                                                                                 [  4%]
test_scan_results/test_macos_inventory.py ..                                                                                                                                                              [  8%]
test_scan_results/test_msu_inventory_msu_feed.py F                                                                                                                                                        [ 10%]
test_scan_results/test_redhat_duplicate_vulns.py ...                                                                                                                                                      [ 17%]
test_scan_results/test_redhat_inventory_redhat_feed.py ....                                                                                                                                               [ 25%]
test_scan_results/test_scan_different_cves.py .........                                                                                                                                                   [ 44%]
test_scan_results/test_scan_nvd_feed.py .............                                                                                                                                                     [ 72%]
test_scan_results/test_scan_providers_and_nvd_feed.py .........                                                                                                                                           [ 91%]
test_scan_results/test_ubuntu_inventory_canonical_feed.py ....                                                                                                                                            [100%]
```